type-definitions/most.d.ts

```diff
@@ -44,7 +44,10 @@ export interface Stream<A> {
   concatMap<B>(f: (a: A) => Stream<B>): Stream<B>;
   mergeConcurrently<B>(concurrency: number): Stream<B>;
   merge(...ss: Array<Stream<A>>): Stream<A>;
+  mergeArray(streams: Array<Stream<A>>): Stream<A>;
   combine<B>(f: (a: A, ...args: Array<any>) => B, ...ss: Array<Stream<any>>): Stream<B>;
+  combineArray<B>(f: (...args: Array<any>) => B, streams: Array<Stream<any>>): Stream<B>;
```

Comment:
> merge and mergeArray have a wrong signature too

I think there is a question here.  Allowing `merge(Stream<A>, Stream<B>): Stream<A | B>` makes it easy to create heterogeneous streams.  One of the goals of types is to provide guidance and help users avoid mistakes.  While heterogeneous streams are *possible*, I'd like to weigh their pros and cons to understand if they are helpful.

Personally, I tend to always use homogeneous streams, and haven't run into a case where I needed a heterogeneous stream.

What are some *concrete* use cases for heterogeneous streams?

Looking around at some strongly typed haskell implementations:

1. Reactive Banana's [unionWith](http://hackage.haskell.org/package/reactive-banana-1.1.0.0/docs/Reactive-Banana-Combinators.html#v:unionWith) is homogeneous.
2. Netwire's [merge, mergeL, and mergeR](http://hackage.haskell.org/package/netwire-5.0.1/docs/Control-Wire-Event.html#v:merge) are homogeneous.
3. FRP.Reactive's [eitherE](http://hackage.haskell.org/package/reactive-0.11.5/docs/FRP-Reactive.html#v:eitherE) is homogeneous in it's output: it *always* produces something of type `Either a b` (it's a subtle but important distinction that it does not produce something of either type 'a' or type 'b').  However, it certainly has a heterogeneous flavor to it.

Thoughts?